lib/Backend/PreLowerPeeps.cpp

```diff
@@ -98,6 +98,14 @@ IR::Instr *Lowerer::PeepShl(IR::Instr *instrShl)
     instrDef->Remove();
 
     IntConstType oldValue = src2->AsIntConstOpnd()->GetValue();
+
+    // Left shift operator (<<) on arm32 is implemented by LSL which doesn't discard bits beyond lowerest 5-bit.
```

Comment:
@LouisLaf @ThomsonTan Thanks for the answers. `x86 masks the shift count to 5bits` is the key difference then. I had confused with the comment above.